### PR TITLE
added a loading message while search results are being fetched 

### PR DESCRIPTION
**Pull request by varenya**
Associated Issue: #4685

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

### Summary of Changes

*  Added a new action creator for updating the status of search result fetching
*  based on the above action, in reducer I have added a new case branch which basically updates the status variable in the reducer.

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] Command-Shift-F opens the "search in files" box
- [x] Enter any text you want to search 
- [x] Press enter
- [x] A "Loading.." message should come up before showing the results.

### CODE REVIEW

File: src/actions/project-text-search.js [hunk 1]
```diff
@@ -34,17 +35,21 @@ export function clearSearchResults() {
   };
 }
 
+export function updateStatus(status: string) {
```
Comment (jasonLaster) [inline src/actions/project-text-search.js]: hmm how about? `updateSearchStatus`

File: src/actions/project-text-search.js [hunk 2]
```diff
@@ -34,17 +35,21 @@ export function clearSearchResults() {
...
+  return { type: "UPDATE_STATUS", status };
+}
+
 export function searchSources(query: string) {
   return async ({ dispatch, getState }: ThunkArgs) => {
     await dispatch(clearSearchResults());
     await dispatch(addSearchQuery(query));
+    dispatch(updateStatus(statusEnum.fetching));
```
Comment (jasonLaster) [inline src/actions/project-text-search.js]: can we set the status to fetching when we set the query or done when we set the results?

Comment (varenya) [reply src/actions/project-text-search.js]: I thought of that at first but this seemed more explicit than your approach

File: src/actions/project-text-search.js (hunk 1 above)
Comment (varenya) [reply src/actions/project-text-search.js]: yeah that sounds better will change it

File: src/reducers/project-text-search.js [hunk 3]
```diff
@@ -22,16 +23,28 @@ export type Search = {
   matches: I.List<any>
 };
 
+export const statusEnum = {
```
Comment (bomsy) [inline src/reducers/project-text-search.js]: nit: can we use `statusType` instead of `statusEnum` feels more intutive

**Comment by bomsy** [comment]
Great work @varenya

**Comment by varenya** [comment]
Thanks everyone :)

**Pull request merged**